### PR TITLE
Consolidate candidate page and source page into a single page; refactor Source.get_if_owned_by

### LINKED ISSUE #1314: Deprecate Candidate page and Source page in favor of single obj page

### LINKED ISSUE #1315: Allow any user to see the obj page of any obj with at least one photometry point they have access to

### PR DESCRIPTION
**Pull request by dannygoldstein**
This pull request does two things:

1. It deprecates the candidates page, which provided only a subset of the functionality of the source page, in favor of a n "obj" page at `/source`. 

2. It refactors the logic for object viewing. Instead of requiring that an object be saved as a source or candidate in one of the user's group to be visible, this PR will let an Obj page be viewed by any user that can see any of its photometry (even if it does not exist as a candidate or source in any of the user's groups). 

Closes #1314. Closes #1315.

### CODE REVIEW

**Comment by acrellin** [comment]
Looking good. We want to allow users to comment on candidates that don't have associated sources?

**Comment by dannygoldstein** [comment]
@acrellin, yeah I think that's fine, as they are just commenting on the Obj.

**Comment by stefanv** [comment]
Looks good so far.

**Comment by dannygoldstein** [comment]
Failures unrelated.

**Review by acrellin — APPROVED** [review]

**Comment by dannygoldstein** [comment]
failures again unrelated

File: static/js/components/Main.jsx.template [hunk 1]
```diff
@@ -87,7 +87,8 @@ const MainContent = ({ root }) => {
 
       <div className={styles.websocket}>
         <WebSocket
-          url={`${window.location.protocol === "https:" ? "wss" : "ws"}://${root}websocket`}
+          url={`${window.location.protocol === "https:" ? "wss" :
```
Comment (stefanv) [inline static/js/components/Main.jsx.template:90]: Accidental commit?

File: static/js/components/CommentList.jsx [hunk 2]
```diff
@@ -150,19 +147,11 @@ const CommentList = ({ isCandidate }) => {
         )}
       </div>
       <br />
-      {!isCandidate && permissions.indexOf("Comment") >= 0 && (
+      {permissions.indexOf("Comment") >= 0 && (
```
Comment (stefanv) [inline static/js/components/CommentList.jsx:150]: How does this PR affect comment viewing permissions on a source?

**Review by stefanv — COMMENTED** [review]
Looks fine to me, Danny.  I suggest you get @dmitryduev to also take a look.

File: static/js/components/Main.jsx.template (hunk 1 above)
Comment (dannygoldstein) [reply static/js/components/Main.jsx.template:90]: Yes, reverted

File: static/js/components/CommentList.jsx (hunk 2 above)
Comment (dannygoldstein) [reply static/js/components/CommentList.jsx:150]: No effect. You still need to be a member of the groups a comment is shared with in order to see a comment, same logic as before.

**Review by stefanv — APPROVED** [review]

**Pull request merged**